### ISSUE TITLE
YAML: `inf` versus `.inf`

### ISSUE THREAD

**Issue by grahamgower** [popsim-consortium/demes-python#130]
The [YAML spec](https://yaml.org/spec/1.2/spec.html#id2804092) says that infinity is spelt `.inf`. It seems `inf` will be interpreted as a string by default. Our strictyaml schema forces the `start_time` field (and others) to be interpreted as a number. So when we put in `inf` (without the leading dot), we get back a python float with value infinity. It's plausible that because we accept this, users would spell it `inf` and then those yaml files would be wrong, but work by accident with `demes`. And for writing yaml files, when `start_time=float("inf")` in python, it seems that our output is `inf` as well, instead of the correct `.inf`.

**Comment by jeromekelleher** [comment]
Good catch, I think it's probably important we enforce it as ``.inf`` for compatibility.

**Comment by grahamgower** [comment]
This is fixed by #132, which uses ruamel directly instead of through strictyaml.
```python
>>> import ruamel.yaml
>>> ruamel.yaml.load("a: inf")
{'a': 'inf'}
>>> ruamel.yaml.load("a: .inf")
{'a': inf}
```
